rbmc-cfam-daemon/test/cfam_fields_test.cpp

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: Apache-2.0
+#include "bmc_cfam.hpp"
+#include "cfam_fields.hpp"
+
+#include <format>
+#include <ranges>
+
+#include <gtest/gtest.h>
+
+using namespace cfam;
+
+TEST(CFAMFields, NoOverLappingFieldsTest)
+{
+    std::array<uint32_t, numScratchPadRegs> regs{};
+
+    // Make sure there aren't any overlapping fields
+    // in BMCCFAM::cfamFields.
+    for (const auto& [field, meta] : BMCCFAM::cfamFields)
+    {
+        auto regIndex = std::to_underlying(meta.reg);
+
+        for (auto i : std::views::iota(meta.start, meta.start + meta.numBits))
+        {
+            if (regs[regIndex] & (0x80000000 >> i))
+            {
+                ADD_FAILURE()
+                    << "Reg " << regIndex << " field "
+                    << static_cast<int>(field)
+                    << " has overlapping bits at offset " << i << "\n";
+            }
+            regs[regIndex] |= (0x80000000 >> i);
+        }
+    }
+
+    std::cout << std::format(
```

Comment:
Suggestion: we could use `std::ranges::fold_left` to avoid repeating the format specifier and index sequentially for each element.